app/scripts/modules/core/src/pipeline/executions/execution/ExecutionMarkerInformationModal.tsx

```diff
@@ -0,0 +1,385 @@
+        });
+      }
+
+      if (childTerminalPipelineStage) {
+        this.getPipelineLink(childTerminalPipelineStage.id, childExecution.id);
+        this.childTerminalPipelineStage = childTerminalPipelineStage;
+      } else {
+        // now that we are complete let's fix up the allExecutions array
+        // we are using allExecutions as a breadcrumb so reverse them then pop the first one since there user is already at the first one
+        this.allExecutions.reverse();
+        this.allExecutions.pop();
+
+        this.setState({
+          pipelineConfigDetails: this.childPipelineConfig || currentPipelineConfig,
+          executionDetails: this.childExecution || currentExecution,
+          failedInApplication: currentStage.masterStage.context.application,
+          link: {
+            application: currentStage.masterStage.context.application,
+            executionId: currentStage.masterStage.context.executionId,
+            stageIndex: this.allExecutions[0].stageIndex,
+          },
+          stageDetails: childTerminalStage || this.childTerminalPipelineStage || currentStage,
+        });
+      }
+    } catch (err) {
+      if (console) {
+        console.error('Error retrieving pipeline execution data.');
+        this.props.onClose();
+      }
+    }
+  };
+
+  private showPipelineGraph = () => {
+    this.setState({ showPipelineGraph: true });
+  };
+
+  private hidePipelineGraph = () => {
+    this.setState({ showPipelineGraph: false });
+  };
+
+  public render(): React.ReactElement<HTMLDivElement> {
+    const { executionDetails, failedInApplication, link, pipelineConfigDetails, stageDetails } = this.state;
+
+    const content = this.state.showPipelineGraph ? (
+      <div className="">
+        {this.allExecutions.map(item => {
+          return (
+            <div key={`${item.execution.id}-${item.execution.name}`} className="execution-graph">
+              {item.execution.application} - {item.execution.name}
+              <PipelineGraph
+                key={item.execution.id}
+                execution={item.execution}
+                onNodeClick={() => {}}
+                viewState={{
+                  activeStageId: item.stageIndex,
+                  activeSubStageId: null,
+                  canTriggerPipelineManually: false,
+                  canConfigure: false,
+                }}
+              />
+            </div>
+          );
+        })}
+      </div>
+    ) : (
+      <div>
+        {!executionDetails ? (
+          <div>
+            <Spinner size="medium" />
+          </div>
+        ) : (
+          <div className="information-details">
+            <div className="pipeline-name bottom-margin">
+              {pipelineConfigDetails ? pipelineConfigDetails.name : '-'}
+            </div>
+            <div style={{ fontWeight: 'bold' }}>PIPELINE</div>
+            <div className="bottom-margin">
+              <div>{`[${executionDetails.authentication.user}] (${executionDetails.user})`}</div>
```

Comment:
For me, this is rendering as `[anonymous] (undefined)` -- let's only render executionDetails.user if present to avoid displaying `undefined`?